libpod/container_internal_linux.go

```diff
@@ -2033,15 +2033,16 @@ func (c *Container) getHosts() string {
 
 		// Do we have a network namespace?
 		netNone := false
-		for _, ns := range c.config.Spec.Linux.Namespaces {
-			if ns.Type == spec.NetworkNamespace {
-				if ns.Path == "" && !c.config.CreateNetNS {
-					netNone = true
+		if c.config.NetNsCtr == "" && !c.config.CreateNetNS {
```

Comment:
Honestly, this is probably safer...

Reply:
Yes we should fix this for the future so that we store the netmode in libpod. But now we are stucked with this forever if we want to support backwards compat.